src/mscorlib/shared/System/ReadOnlySpan.cs

```diff
@@ -321,5 +321,45 @@ public T[] ToArray()
         /// Returns a 0-length read-only span whose base is the null pointer.
         /// </summary>
         public static ReadOnlySpan<T> Empty => default(ReadOnlySpan<T>);
+
+        /// <summary>Gets an enumerator for this span.</summary>
+        public Enumerator GetEnumerator() => new Enumerator(this);
+
+        /// <summary>Enumerates the elements of a <see cref="ReadOnlySpan{T}"/>.</summary>
+        public ref struct Enumerator
```

Comment:
Consider moving to separate file? Is nesting the Enumerator struct within ReadOnlySpan necessary?

Reply:
It's not strictly necessary, but it's the pattern we follow for all such Enumerator structs across all such types in the framework.

Reply:
Fair enough. I saw some examples with nested structs:
https://github.com/dotnet/corefx/blob/master/src/System.Reflection.Metadata/src/System/Reflection/Metadata/PortablePdb/HandleCollections.Debug.cs#L51

Reply:
Right, pretty much all of our enumerator structs are nested, e.g.
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Collections/Generic/Dictionary.cs#L885
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Collections/Generic/List.cs#L1173
https://github.com/dotnet/corefx/blob/master/src/System.Collections/src/System/Collections/Generic/Queue.cs#L395
etc.